### PR TITLE
Client.hh: Added a loop around initial client connection to allow the…

### LINKED ISSUE #86: Client-Connection-Retry
Currently the client connection can fail on a race condition of both the server and client sims come up at the same time. This throws a segfault. We propose adding the connection inside of a loop to allow for a few retries before throwing an exception.

### PR DESCRIPTION
**Pull request by cdforema**
… socket connection to attempt to be open several times in case it fails initially due to race conditions

### CODE REVIEW

File: include/idf/Client.hh [hunk 1]
```diff
@@ -172,6 +195,9 @@ class Client : public Manageable {
 
     private:
 
+    /** Number of times to retry establishing the initial connection */
+    int retryLimit;
```
Comment (jwpetti1) [inline include/idf/Client.hh]: This is generating a new compiler warning on macOs

> IDF/include/idf/Client.hh:175:9: warning: field 'serverPort' will be initialized after field 'retryLimit' [-Wreorder-ctor]
        serverPort(port),
        ^
IDF/source/idf/RemoteSpaceNavigator.cpp:23:5: note: in instantiation of member function 'idf::Client<idf::SpaceNavigator, idf::SpaceNavigatorCommands>::Client' requested here
    Client<SpaceNavigator, SpaceNavigatorCommands>(spaceNavigator, hostName, hostPort) {}
    ^
1 warning generated.

Moving this declaration below `serverPort` should fix

**Review by jwpetti1 — APPROVED** [review]

**Review by prkunz — APPROVED** [review]
Looks good to me. Although I'm not setup to test a remote connection

File: include/idf/Client.hh (hunk 1 above)
Comment (prkunz) [reply include/idf/Client.hh]: is the Mac compiler complaining about alphabetical order if member definitions?

**Comment by prkunz** [comment]
closes #86

**Pull request merged**